cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -367,12 +367,12 @@ func addDeviceInstanceAndProtocol(device *v1alpha2.Device, deviceProfile *types.
 // deviceAdded creates a device, adds in deviceManagers map, send a message to edge node if node selector is present.
 func (dc *DownstreamController) deviceAdded(device *v1alpha2.Device) {
 	dc.deviceManager.Device.Store(device.Name, device)
-	if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
+	if nodeID := getDeviceNode(device); len(nodeID) != 0 {
```

Comment:
Let's use `nodeName` instead of `nodeID` in all of this PR,  `nodeID` may be confused with `node UUID`

Reply:
ok, done